[CI] DISIAccumulatorTests testRandom failing

**Build Scans:**
- [elasticsearch-intake #42531 / part1](https://gradle-enterprise.elastic.co/s/uum5eczokn77e)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.tsdb.DISIAccumulatorTests.testRandom" -Dtests.seed=955220E412F1A940 -Dtests.locale=en-AG -Dtests.timezone=Etc/GMT-8 -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.tsdb.DISIAccumulatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRandom'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testRandom
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -292,9 +292,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT
   method: testStress
   issue: https://github.com/elastic/elasticsearch/issues/152671
-- class: org.elasticsearch.index.codec.tsdb.DISIAccumulatorTests
-  method: testRandom
-  issue: https://github.com/elastic/elasticsearch/issues/152677
 - class: org.elasticsearch.repositories.azure.AzureBlobContainerRetriesTests
   method: testWriteLargeBlob
   issue: https://github.com/elastic/elasticsearch/issues/152914

--- a/server/src/test/java/org/elasticsearch/index/codec/tsdb/DISIAccumulatorTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/tsdb/DISIAccumulatorTests.java
@@ -19,6 +19,7 @@ import org.apache.lucene.util.BitSet;
 import org.apache.lucene.util.BitSetIterator;
 import org.apache.lucene.util.FixedBitSet;
 import org.apache.lucene.util.SparseFixedBitSet;
+import org.apache.lucene.util.Version;
 import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.test.ESTestCase;
 
@@ -398,7 +399,18 @@ public class DISIAccumulatorTests extends ESTestCase {
     private void doTestRandom(Directory dir) throws IOException {
         Random random = random();
         final int maxStep = TestUtil.nextInt(random, 1, 1 << TestUtil.nextInt(random, 2, 20));
-        final int numDocs = TestUtil.nextInt(random, 1, Math.min(100000, (Integer.MAX_VALUE - 1) / maxStep));
+        // SparseFixedBitSet.blockCount() has an assertion that overflows for lengths >= 2_147_479_553
+        // in Lucene 10.5 (fixed in https://github.com/apache/lucene/pull/14922). Cap numDocs so that
+        // numDocs * maxStep + 99 (the max possible maxDoc) stays below that threshold.
+        // TODO: remove this cap after upgrading Lucene past 10.5.
+        assert Version.LUCENE_10_5_0.onOrAfter(Version.LATEST)
+            : "Lucene has been upgraded past 10.5; remove the SparseFixedBitSet range cap in doTestRandom";
+        final int sparseFixedBitSetSafeCap = (int) (2_147_479_453L / maxStep);
+        final int numDocs = TestUtil.nextInt(
+            random,
+            1,
+            Math.min(Math.min(100000, (Integer.MAX_VALUE - 1) / maxStep), sparseFixedBitSetSafeCap)
+        );
         BitSet docs = new SparseFixedBitSet(numDocs * maxStep + 1);
         int lastDoc = -1;
         for (int doc = -1, i = 0; i < numDocs; ++i) {

